### ISSUE TITLE
rfuchs/socket-rework branch status

### ISSUE THREAD

**Issue by egreenmachine** [sipwise/rtpengine#156]
Hi,

I am working with @etamme at @OnSIP on some recording stuff with rtpengine. We are trying to get the rfuchs/socket-rework branch running on our boxes to start looking into recording RTP streams. Both endpoints are WebRTC. The rtpengine is segfaulting on the answer. Is this branch expected to compile and run cleanly in its current state? If so, any insight into our segfault?

Also what are the required versions of glib and kernel to run this branch?

Thanks!

~~~
#0  0x0000000000412a4f in call_stream_address46 (o=0x7fa092199f70 "\300\241\031\222\240\177", ps=0x7fa08801ef10, format=SAF_NG, len=
    0x7fa092199fbc, ifa=0x0) at call.c:2070
#1  0x000000000041bf92 in replace_network_address (chop=0x7fa088024d00, address=0x7fa088008368, ps=0x7fa08801ef10, flags=0x7fa09219a1c0)
    at sdp.c:1483
#2  0x000000000041fd09 in sdp_replace (chop=0x7fa088024d00, sessions=<value optimized out>, monologue=<value optimized out>, 
    flags=<value optimized out>) at sdp.c:1931
#3  0x0000000000428b2a in call_offer_answer_ng (input=<value optimized out>, m=0x14580d0, output=0x7fa08800fc18, 
    opmode=<value optimized out>, addr=0x7fa088024d00 "\020\243\031\222\240\177", sin=0x0) at call_interfaces.c:707
#4  0x000000000041bb0b in control_ng_incoming (obj=0x145a000, buf=<value optimized out>, sin=0x7fa0921aa4f0, addr=
    0x7fa0921aa4b0 "38.109.82.162:35020", ul=0x145a0a0) at control_ng.c:150
#5  0x000000000041ad23 in udp_listener_incoming (fd=<value optimized out>, p=0x14538a0, x=<value optimized out>) at udp_listener.c:56
#6  0x000000000040fcae in poller_poll (p=0x14550f0, timeout=100) at poller.c:348
#7  0x000000000040fcd8 in poller_loop (d=0x14550f0) at poller.c:502
#8  0x000000000041042f in thread_detach_func (d=0x144fab0) at aux.c:143
#9  0x0000003c822079d1 in start_thread () from /lib64/libpthread.so.0
#10 0x0000003c81ae88fd in clone () from /lib64/libc.so.6
~~~

**Comment by rfuchs** [comment]
The code in this branch has full functionality, but as you've found out there might still be a few bugs lurking here and there. The required libs and kernel versions are the same as for master.

Can you please post the log from this call leading up to the crash?

**Comment by egreenmachine** [comment]
So I did a bit more investigating into this since I first posted. It appears to be only happening with 2 WebRTC clients on the same machine trying to do video. I have rtpengine set to force itself to relay between the 2. Doing video between clients on different machines appears to work fine. Doing audio with 2 clients on the same machine appears to work fine. 

Log: https://gist.github.com/egreenmachine/8b388a513471f64666b2
Core Dump File: https://drive.google.com/file/d/0B3M8hSjrBv23bW5sazVaTW1RSlU/view?usp=sharing
